internal/plugincommon/consensus/threshold.go

```diff
@@ -0,0 +1,47 @@
+package consensus
+
+// Threshold is the minimum number of observations required for a given chain.
+type Threshold uint
+
+// MultiThreshold is an interface that maps a chain selector to a threshold.
+type MultiThreshold[T any] interface {
```

Comment:
Small comment on the generics: this says `T any` but can this actually be done with anything that is not `comparable`?

Reply:
Actually doesn't need to be `comparable`, since `ConstantThreshold` doesn't need to make any comparisons. Nice!

Reply:
Yup, only the key needs to be `comparable`